js/resources_test.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import { test, testPerm, assert, assertEqual } from "./test_util.ts";
+import * as deno from "deno";
+
+test(function resourcesStdio() {
+  const res = deno.resources();
+
+  assertEqual(res[0], "stdin");
+  assertEqual(res[1], "stdout");
+  assertEqual(res[2], "stderr");
+});
+
+testPerm({ net: true }, async function resourcesNet() {
+  const addr = "127.0.0.1:4501";
+  const listener = deno.listen("tcp", addr);
+
+  listener.accept().then(async conn => {
+    const res = deno.resources();
+    // besides 3 stdio resources, we should have additional 3 from listen(), accept() and dial()
+    assertEqual(Object.keys(res).length, 6);
+    assertEqual(Object.values(res).filter(r => r === "tcpListener").length, 1);
+    assertEqual(Object.values(res).filter(r => r === "tcpStream").length, 2);
+
+    conn.close();
+    listener.close();
+  });
+
+  const conn = await deno.dial("tcp", addr);
+  const buf = new Uint8Array(1024);
+  await conn.read(buf);
+  conn.close();
```

Comment:
It's possible that the callback doesn't get called at all. Please add a counter which gets incremented in the callback, and check here
`assertEqual(counter, 1)`

Also you can remove ` await conn.read(buf);` - I think that has no effect here.

Reply:
You were right about that, I reworked it to use `await`, let me know if it looks good.